
<launch>

  <param name="/use_sim_time" value="true" />

  <!--- static TF -->
  <node pkg="tf" type="static_transform_publisher" name="sensor_in_chassis"  args="0 0 1.3 0 0 0 /chassis /velodyne 10" />
  <node pkg="tf" type="static_transform_publisher" name="basescan_in_chassis"  args="0 0 1.3 0 0 0 /chassis /base_scan 10" />

  <!-- ROS Node -->
  <node name="advanced_segmentation" pkg="advanced_segmentation" type="advanced_segmentation_node" output="screen">
    <rosparam command="load" file="$(find advanced_segmentation)/launch/advanced_params.yaml"/>
    <param name="sub_cloud_topic" value="/velodyne_points" />
    <param name="pub_cloud_topic" value="/AS_segmted_cloud" />
    <param name="pub_rangeimage_topic" value="/AS_range_image" />
    <param name="pub_extractedlines_topic" value="/AS_extracted_lines" />
  </node>


  <!-- ROS Rviz -->
  <node name="rviz_AdvancedSegmentation" pkg="rviz" type="rviz" required="true"
    args="-d $(find advanced_segmentation)/launch/advanced_segmentation.rviz" />


  <!-- 数据集|地库环境 -->
  <arg name="bag_filename" default="/home/wgh/aaa_bags/2021-04-27-B1-03-fromElevator.bag"/>

  <!-- 数据集|室外环境：北广场/高动态/南广场 -->
  <!-- <arg name="bag_filename" default="/home/wgh/aaa_bags/2021-04-27-B1-03-fromElevator.bag"/> -->


  <!-- ROS Bag Play -->
  <node pkg="rosbag" type="play" name="rosbag_play" args="--clock -s 0 -r 1 $(arg bag_filename)"/>

</launch>
